plugins/dialog/plugin.js

```diff
@@ -905,7 +905,8 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 				CKEDITOR.dialog._.currentTop = this;
 				this._.parentDialog = null;
 				showCover( this._.editor );
-			} else {
+			} else if ( CKEDITOR.dialog._.currentTop !== this ) {
+				// Do the magic with z-indexes only if the current dialog is not already visible (#3638).
```

Comment:
I'm not 100% sure about this comment. I mean the part with zIndex. The entire `else...if` is prevented `if the current dialog is not already visible`. So besides zIndex operation we also have assignments to `parentDialog` and `currentTop`.

We could leave just the part with issue reference at the top, and move the zIndex part just before zIndex operation?